test/collection_test.js

```diff
@@ -62,6 +62,21 @@ describe("Collection", () => {
     };
   }
 
+  function createKeyListIdSchema() {
+    return {
+      generate(record) {
+        let id = [];
+        for (let key in record) {
+          id.push(key);
+        }
+        return id;
```

Comment:
I personnally find this is a bit weird... the usage example you had in the docs would make a lot more sense IMO :)